### PR TITLE
Project Manager Fixes for Linux

### PR DESCRIPTION
**Pull request by spham-amzn**
- Cherry-pick auto clang detection for Linux 
- Fix error with EngineFinder using the wrong path to locate project.json
- Simplified and expanded clang detection
- Remove forcing clang-12 for builds and will rely on the new cmake detection of clang by default

### CODE REVIEW

File: Templates/DefaultProject/Template/cmake/EngineFinder.cmake [hunk 1]
```diff
@@ -13,8 +13,8 @@
 include_guard()
 
 # Read the engine name from the project_json file
-file(READ ${CMAKE_CURRENT_LIST_DIR}/project.json project_json)
-set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_LIST_DIR}/project.json)
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/project.json project_json)
```
Comment (amznestebanpapp) [inline Templates/DefaultProject/Template/cmake/EngineFinder.cmake:16]: thanks for catching this!

**Review by amznestebanpapp — APPROVED** [review]

File: Code/Tools/ProjectManager/Platform/Linux/ProjectUtils_linux.cpp [hunk 2]
```diff
@@ -39,16 +37,13 @@ namespace O3DE::ProjectManager
...
-                AZ_Assert(clangCompilers.length()==2, "Invalid clang compiler pair specification");
-
-                auto whichClangResult = ProjectUtils::ExecuteCommandResult("which", QStringList{clangCompilers[0]}, QProcessEnvironment::systemEnvironment());
-                auto whichClangPPResult = ProjectUtils::ExecuteCommandResult("which", QStringList{clangCompilers[1]}, QProcessEnvironment::systemEnvironment());
+                auto whichClangResult = ProjectUtils::ExecuteCommandResult("which", QStringList{QString("clang-%1").arg(supportClangVersion)}, QProcessEnvironment::systemEnvironment());
+                auto whichClangPPResult = ProjectUtils::ExecuteCommandResult("which", QStringList{QString("clang++-%1").arg(supportClangVersion)}, QProcessEnvironment::systemEnvironment());
                 if (whichClangResult.IsSuccess() && whichClangPPResult.IsSuccess())
                 {
-                    return AZ::Success(supportClangCommand);
+                    return AZ::Success(QString("clang-%1").arg(supportClangVersion));
```
Comment (lumberyard-employee-dm) [inline Code/Tools/ProjectManager/Platform/Linux/ProjectUtils_linux.cpp:46]: You can remove this function now, that it ProjectManager isn't trying to do compiler detection.

Comment (spham-amzn) [reply Code/Tools/ProjectManager/Platform/Linux/ProjectUtils_linux.cpp:46]: I couldn't, its a common function that is implemented for all platforms. Project Manager uses it to make sure you have a compatible compiler before you even attempt to generate a project. 

The difference now is that the detected compiler isnt used for the cmake command line anymore.

Comment (lumberyard-employee-dm) [reply Code/Tools/ProjectManager/Platform/Linux/ProjectUtils_linux.cpp:46]: Alright

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**